### ISSUE TITLE
RC1 Error 0x80091007: Hash mismatch for path: DotNetVersionManager_x64, expected: BAEF7329894C48023E4E0F5F7D5C84E2243309F8, actual: FB0D8F2A91BEAC7991DC7A3FC13DA5F3D9E513C0

### ISSUE THREAD

**Issue by Anderman** [aspnet/Home#1085]
I got this error when running from http://get.asp.net/

![image](https://cloud.githubusercontent.com/assets/1858745/11246786/7749c144-8e19-11e5-8d5b-86c366b06ba7.png)

**Comment by DamianEdwards** [comment]
This is a known issue. Delete any previous MSIs for DNVM and ASP.NET VS tooling from the folder you downloaded the RC1 installer to. We're working to fix this issue.

**Comment by Eilon** [comment]
Delete this folder: `C:\ProgramData\Package Cache\{C5A4ABA3-1ABA-3EF8-B2D5-C3FA37F59738}v1.0.10609.0\packages\WPT`

We're working on a fix. Sorry about that!

**Comment by sirkirby** [comment]
fixed worked for me and team...thanks!

**Comment by joeloff** [comment]
@sirkirby Glad to hear it worked. Would you mind sending me a copy of the log from the failed install. It should be in %temp% and have a name similar to dd_ASPNET5__20151118084006.log. You can send it directly to me at joeloff at Microsoft dot com

**Comment by sirkirby** [comment]
@joeloff here you go

[dd_ASPNET5__20151118102506.txt](https://github.com/aspnet/Home/files/38097/dd_ASPNET5__20151118102506.txt)

**Comment by Loghorn** [comment]
I've the same issue but the delete folder fix proposed by @Eilon doesn't work (I don't have that folder at all). Here is my log
[dd_ASPNET5__20151118181605.txt](https://github.com/aspnet/Home/files/38116/dd_ASPNET5__20151118181605.txt)

**Comment by OlegKi** [comment]
I have the same issue, but I can see
>3A70:13A8][2015-11-18T18:09:33]e000: Error 0x80091007: Hash mismatch for path: C:\ProgramData\Package Cache\.unverified\DotNetVersionManager_x64, expected: BAEF7329894C48023E4E0F5F7D5C84E2243309F8, actual: **FB0D8F2A91BEAC7991DC7A3FC13DA5F3D9E513C0**

in my log file instead of
>[9FE40:B215C][2015-11-18T10:25:09]e000: Error 0x80091007: Hash mismatch for path: C:\ProgramData\Package Cache\.unverified\DotNetVersionManager_x64, expected: BAEF7329894C48023E4E0F5F7D5C84E2243309F8, actual: **8A5128FFAC8D3E1C92AD6252BADD9B8CA10F07A8**

in the log of **sirkirby**.
**Deleting of the folder fix proposed by @Eilon doesn't work for me too**
I include my full log
[dd_ASPNET5__20151118180803.log.txt](https://github.com/aspnet/Home/files/38115/dd_ASPNET5__20151118180803.log.txt)
I will continue to investigate myself, but I want to share the log with other first.

**Comment by sebastienros** [comment]
Moving the downloaded MSI on the deskptop worked for me. Apparently the fact that there are other MSIs in the same folder is the issue.

**Comment by Loghorn** [comment]
Moving the downloaded file to an empty folder fixed the issue for me.
Thanks @sebastienros

**Comment by russdaygh** [comment]
@sebastienros moving worked for me too.

**Comment by Anderman** [comment]
Yep for mee too

**Comment by joeloff** [comment]
Yup, if you download the EXE and save it, e.g. c:\users\bob\downloads and have installers from the previous Beta releases in the same downloads folder you will hit the issue.

Workaround is to make sure there are none of the previous MSI files from the Beta releases in the folder where the EXE is downloaded.

Alternative is to run the EXE directly when prompted by the browser instead of saving it.

**Comment by OlegKi** [comment]
@joeloff & @sebastienros : Thank you! Moving of `AspNet5.ENU.RC1.exe` in separate empty folder before starting solves the issue.

**Comment by herecydev** [comment]
@joeloff :+1: This was what made it work for me.

**Comment by avanderhoorn** [comment]
I got this same issue and the moving to a new folder fix worked for me.

**Comment by muratg** [comment]
FYI, a fix to this issue is in progress.

**Comment by KantJoe** [comment]
i hava 2 pc with different .net develop environment,and one has occur this error,but another not.perhaps,the problem may be not easy.
the pc with vs2013,vs2015express,vs2015comunity can installs DotNetVersionManager-x64for vs2015express success,than i install rc1 for vs2015community.success too.then i install rc1 for my 2nd pc with vs2013,vs2015comunity but failed.hope these info can help u guys figure out

**Comment by joeloff** [comment]
@KantJoe Have you tried moving the EXE for RC1 to a separate folder and run it from there.

**Comment by KantJoe** [comment]
@joeloff yes，i did,but failed and i tried to delete temporary files,unistall web tool 4.6 and dotnetversionmanager64 ,at last succeeded by step: download dotnetversionmanager64 to rc1 folder

**Comment by luckyhappy** [comment]
I tried moving the AspNet5.ENU.RC1_Update1.exe to a new folder,and delete this folder: C:\ProgramData\Package Cache\{C5A4ABA3-1ABA-3EF8-B2D5-
C3FA37F59738}v1.0.10609.0\packages\WPT; still install fail.
[dd_ASPNET5__20151203205259.txt](https://github.com/aspnet/Home/files/50829/dd_ASPNET5__20151203205259.txt)

**Comment by Eilon** [comment]
@joeloff

**Comment by joeloff** [comment]
Looks like the setup blocker (to warn you if VS is open) was cached on disk, but it failed to run on the machine. I'm following up on this.

**Comment by joeloff** [comment]
@luckyhappy can you see if the folder C:\ProgramData\Package Cache\da880e12-47d7-30ce-b7cc-8b2ca3aa0f06\ exists and if so, delete it and rerun the installer?

**Comment by luckyhappy** [comment]
@joeloff,not exists

**Comment by joeloff** [comment]
@luckyhappy A few things to try. Could you delete the EXE and download it again and run it. The application it's trying to run during the install is embedded in the RC1 bundle, but I want to rule out any potential corruption in the embedded container. Everything in the logs seem to indicate that it failed to get the embedded file extracted and put on disk. Something else that might interfere is anti-virus software, but we haven't seen anything around this.

**Comment by luckyhappy** [comment]
@joeloff ,Success!  I reinstall it again, although I installed it at least 5 times before  and failed,
Thanks patience to help.

**Comment by duracellko** [comment]
I found in log:
```
[0B38:0B34][2015-12-04T22:10:40]w120: Detected partially cached package: DotNetVersionManager_x64, invalid payload: DotNetVersionManager_x64, reason: 0x80070002
[0B38:0B34][2015-12-04T22:10:40]i103: Detected related package: {C5A4ABA3-1ABA-3EF8-B2D5-C3FA37F59738}, scope: PerMachine, version: 1.0.10609.0, language: 0 operation: MajorUpgrade
[0B38:0B34][2015-12-04T22:10:40]i103: Detected related package: {F7BBF232-D4A1-3C5C-B131-69933F5F2D6B}, scope: PerMachine, version: 14.1.11106.0, language: 0 operation: MajorUpgrade
```
So I deleted folders:
```
C:\ProgramData\Package Cache\{C5A4ABA3-1ABA-3EF8-B2D5-C3FA37F59738}*
```
and
```
C:\ProgramData\Package Cache\{F7BBF232-D4A1-3C5C-B131-69933F5F2D6B}*
```

asterisk is a wildcard. I don't remember exact folder names, cause I deleted those :D
and the installation works after that

**Comment by guganscode** [comment]
Install the latest visual c++ package, then cleanup all the files under temp folder, move all the files from C:\Users\<user>\AppData\Local\Package Cache, It should work now.

**Comment by Brij-Mishra** [comment]
There is another issue while installing the same tool which is logged here around two weeks back. It is acknowledged but no update since last 10 days. Any update please?  https://connect.microsoft.com/VisualStudio/Feedback/Details/2112077

**Comment by jvsca** [comment]
FYI: On Windows Server 2012 R2, need to turn off IE Enhanced Security Configuration to run a successful install.

**Comment by jakkavrao** [comment]
Hi, im still getting the same error. I tried all fixes suggested above. no luck
[dd_ASPNET5__20160216044418.TXT](https://github.com/aspnet/Home/files/131604/dd_ASPNET5__20160216044418.TXT)

**Comment by ThangaBalajiS** [comment]
Hi, I tried all the methods mentioned above, still couldn't get that.. please help me fix this, thanks in advance 

[ASP.NEt.txt](https://github.com/aspnet/Home/files/141664/ASP.NEt.txt)

**Comment by edelweiss121165** [comment]
Hi,
I am still experiencing the same issue on Windows 7 Professional. Tried many times all of the above but it still getting the same error.
If somebody could give me some ideas of what could be wrong. 
File AspNet5.ENU.RC1_Update1_KB3137909.EXE is on my desktop but I am still getting error 0x80091007: 
Hash mismatch for path: C:\ProgramData\Package Cache\.unverified\DotNetVersionManager_x86...
Please see the log file attached. Thank you in advance.

**Comment by guangdi-fiserv** [comment]
I moved in new  location but I still have "[20EC:1C78][2016-04-25T10:30:34]e000: Error 0x80091007: Hash mismatch for path: C:\ProgramData\Package Cache\.unverified\DotNetVersionManager_x64, expected: 7D5ED0F9A713E9684475D43394F6082C328FE879, actual: DA39A3EE5E6B4B0D3255BFEF95601890AFD80709
[20EC:1C78][2016-04-25T10:30:34]e000: Error 0x80091007: Failed to verify hash of payload: DotNetVersionManager_x64
[20EC:1C78][2016-04-25T10:30:34]e310: Failed to verify payload: DotNetVersionManager_x64 at path: C:\ProgramData\Package Cache\.unverified\DotNetVersionManager_x64, error: 0x80091007. Deleting file."
